assets/js/swedbank-pay-design-guide-theme.js

```diff
@@ -20,54 +20,83 @@
 
 // Initialize sidebar navigation scroll activation
 (function () {
-    var headings = document.querySelectorAll("h2");
-    var tocLinks = document.querySelectorAll("nav.doc-toc ul li a");
+    var headings = document.querySelectorAll(".doc-container h2");
+    var tocLinks = document.querySelectorAll("nav.sidebar-nav .nav-subgroup.active .nav-leaf");
+
+    if (tocLinks.length === 0) {
+        tocLinks = document.querySelectorAll("nav.sidebar-nav .nav-group.active .nav-leaf");
+    }
 
     var getPosition = function (el) {
         if (el) {
             var bodyRect = document.body.getBoundingClientRect();
             var elemRect = el.getBoundingClientRect();
-
+            
             return elemRect.top - bodyRect.top;
         }
-
+        
         return null;
     };
-
+    
     window.addEventListener("scroll", function () {
-        var buffer = document.body.clientHeight * 0.1;
-        var currentPos = window.pageYOffset + buffer;
-
-        // TODO: Probably a stupid way to compute "how far left can we scroll until
-        //       we reach the bottom of the page", but it seems to work.
-        var scrollDistanceFromBottom = document.documentElement.scrollHeight
-            - document.documentElement.scrollTop
-            - document.body.clientHeight
-            - buffer;
+        if (tocLinks.length > 0) {
+            var activeLeaf = document.querySelector("nav.sidebar-nav .nav-leaf.active");
+            var buffer = document.body.clientHeight * 0.1;
+            var currentPos = window.pageYOffset + buffer;
+    
+            // TODO: Probably a stupid way to compute "how far left can we scroll until
+            //       we reach the bottom of the page", but it seems to work.
+            var scrollDistanceFromBottom = document.documentElement.scrollHeight
+                - document.documentElement.scrollTop
+                - document.body.clientHeight
+                - buffer;
+                
+                
+                activeLeaf && activeLeaf.classList.remove("active");
+                
+                
+            if (scrollDistanceFromBottom > 0) {
+                const scrollNumber = [...headings].filter((heading) => getPosition(heading) <= currentPos).length - 1;
 
-        if (scrollDistanceFromBottom <= 0) {
-            for (var link of tocLinks) {
-                link.parentElement.classList.remove("active");
+                scrollNumber >= 0 && tocLinks[parseInt(scrollNumber, 10)].classList.add("active");

```

Comment:
The `parseInt` is due to Codacy complaining about "Generic Object Injection Sink", which seems to only be an issue if the variable used as key gets its value from an external source (a user input for instance). Just made a work around for this, as scrollNumber is an internal `const ` and does not seem to introduce any security risks